Kinvey/Kinvey/RealmCache.swift

```diff
@@ -66,16 +68,41 @@ internal class RealmCache<T: Persistable>: Cache<T> where T: NSObject {
         return realmResults
     }
     
-    fileprivate func newInstance<P: Persistable>(_ type: P.Type) -> P {
+    fileprivate func newInstance<P:Persistable>(_ type: P.Type) -> P {
         return type.init()
     }
-    
-    override func detach(_ entity: T) -> T {
+
+    fileprivate func detach(_ entity: Object, props: [String]) -> Object {
```

Comment:
@tejasranade Is there a reason that `detach(entity, props)` is `fileprivate` while the `detach(entity, query)` is `public`?  Also, because `detach(entity, props)` is now `fileprivate`, does it have to be deprecated?